src/xdsl/rewriter.py

```diff
@@ -0,0 +1,199 @@
+import inspect
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import List, Optional, Callable, Dict
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import Operation, SSAValue, Region, Block
+
+
+@dataclass(init=False)
```

Comment:
What is the benefit of having a `dataclass `with `init=False`?

Reply:
True, I'll remove it, since it only generates equality and representation, which we don't really care here

Reply:
I actually kept it, but I added init back.